protocol/binary_test.go

```diff
@@ -376,6 +381,24 @@ func TestStruct(t *testing.T) {
 	checkEncodeDecode(t, wire.TStruct, tests)
 }
 
+func TestStructBeginAndEndEncode(t *testing.T) {
+	var (
+		streamBuff bytes.Buffer
+		err        error
+	)
+
+	// Encode with Streaming protocol
+	w := binary.BorrowStreamWriter(&streamBuff)
+	err = w.WriteStructBegin()
+	assert.NoError(t, err)
+	err = w.WriteStructEnd()
+	assert.NoError(t, err)
```

Comment:
optional: can simplify:
```suggestion
	require.NoError(t, w.WriteStructBegin())
	require.NoError(t, w.WriteStructEnd())
```